### DISCUSSION TITLE
Where are virtual foreign keys and diagrams persisted?

### DISCUSSION THREAD

**Discussion by jsperson** [dbeaver/dbeaver#8772] [Q&A] [answered]
#### System information: 
- Windows 10 - fully updated
- DBeaver version - 7.0.0.202003081756

#### Connection specification:
- Snowflake

#### Describe your question:
Where is the meta-data for virtual foreign keys and diagrams stored?

**Comment by kseniiaguzeeva** [comment, marked as answer]
To see a table/view virtual foreign key in DBeaver, please open a table/view in DBO Editor>Properties>Virtual tab
![dbeaver_vEeyfseCDM](https://user-images.githubusercontent.com/51405061/82670651-363f8e00-9c46-11ea-8c92-983a60ee042f.png)
To see saved custom ERD in DBeaver, open Projects>ER Diagrams
To see saved custom ERD file on you computer, select properties in ERD context menu and see the location.
![kji5Yle4jo](https://user-images.githubusercontent.com/51405061/82670851-96cecb00-9c46-11ea-8f26-75af9329991c.png)

Please reopen the ticket if you have additional questions.

**Comment by pinkfrog9** [comment]
I have a two tables `foo` and `bar`.  `foo` has a column `bar_id` which points to `bar`, a vritual foreign key.

I wonder how can dbeaver automatically show the link between `foo` and `bar` when it shows the ER diagram of `foo`?

datagrip has this feature by default so I wonder how's the situation with dbeaver.

Thanks
